va/va.go

```diff
@@ -251,99 +251,85 @@ type verificationRequestEvent struct {
 	Error             string `json:",omitempty"`
 }
 
+// ipError is a sentinel error type used to pass though the IP address of the
+// remote host when an error occurs during HTTP-01 and TLS-ALPN domain
+// validation.
+type ipError struct {
+	ip  net.IP
+	err error
+}
+
+// Unwrap returns the underlying error.
+func (i ipError) Unwrap() error {
+	return i.err
+}
+
+// Error returns a string representation of the error.
+func (i ipError) Error() string {
+	return i.err.Error()
+}
```

Comment:
It's weird not to include the IP here. I would expect `fmt.Sprintf("%s: %s", i.ip, i.err)`. Of course that would be duplicative with some of the logic below, but I think that's fine. You never wind up calling `.Error()` on this type, but it's nice for it to return the correct thing if, for instance, we print it during debugging.